tests/TestCase/Routing/Route/EntityRouteTest.php

```diff
@@ -23,6 +23,34 @@
  */
 class EntityRouteTest extends TestCase
 {
+    /**
+     * test that route keys take precedence to object properties.
+     *
+     * @return void
+     */
+    public function testMatchRouteKeyPrecedence()
+    {
+        $entity = new Article([
+            'category_id' => 2,
+            'slug' => 'article-slug'
+        ]);
+
+        $route = $route = new EntityRoute(
+            '/articles/:category_id/:slug',
+            [
+                '_name' => 'articlesView',
+            ]
+        );
+
+        $result = $route->match([
+            'slug' => 'other-slug',
+            '_entity' => $entity,
+            '_name' => 'articlesView'
+        ]);
+
+        $this->assertEquals('/articles/2/other-slug', $result);
```

Comment:
needs another test for `Route defaults`

Reply:
Route defaults currently cannot overlap routing keys. That feature will be implemented late in 3.6 though.